### PR TITLE
Implement dialogs entry points in api

### PR DESCRIPTION
**Pull request by javierluraschi**
Making the following functions available through this package:

- `showDialog(title, message, url = NULL)`
- `updateDialog(...)`
- `showQuestion(title, message, ok = NULL, cancel = NULL)`
- `showPrompt(title, message, default = NULL)`
- `writePreference(name, value)`
- `readPrefernce(name, default)`

Dependent on: https://github.com/rstudio/rstudio/pull/967

### CODE REVIEW

File: NAMESPACE [hunk 1]
```diff
@@ -35,14 +35,20 @@ export(modifyRange)
 export(navigateToFile)
 export(previewRd)
 export(primary_selection)
+export(readPreference)
```
Comment (jjallaire) [inline NAMESPACE]: I'd vote to not expose these.

File: R/dialogs.R [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+#' Currently, the only dialog with support for this action is the
+#' New Connection dialog in which the code preview can be
+#' updated through this API.
+#'
+#' \preformatted{
+#'     updateDialog(code = "con <- NULL")
+#' }
+#'
+#' @export
+updateDialog <- function(...) {
```
Comment (jjallaire) [inline R/dialogs.R]: Maybe `updateActiveDialog` ?

File: R/dialogs.R [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+#'   the Cancel button.
+#'
+#' @details
+#'
+#' @export
+showQuestion <- function(title, message, ok = NULL, cancel = NULL) {
+  callFun("showQuestion", title, message, ok, cancel)
+}
+
+#' Read Preference
```
Comment (jjallaire) [inline R/dialogs.R]: Okay, I get why these are exposed now (dialog UI remembering prefs). I think it's okay to expose these but we should "namespace" them sort of like we did here for persistent values: https://github.com/rstudio/rstudio/commit/47ae3dc69090ccd3e31c42a4d63b0b5c3fc8e9f8

**Comment by jjallaire** [comment]
We usually also note the version of RStudio where APIs were introduced within the `@details` section of the help topics (assuming your changes in rstudio get merged today that can just be the next increment of the daily build version).

File: R/dialogs.R (hunk 3 above)
Comment (javierluraschi) [inline R/dialogs.R]: @jjallaire they are already namespaced with `rstudioapi_`: https://github.com/rstudio/rstudio/pull/967/files#diff-ede85aaa4cad3424f4cae655c5408f80R384 - Let me know if we prefer a different namespace and I'll make the change.

**Comment by javierluraschi** [comment]
@jjallaire thanks for the merge, just pushed the note with RStudio version to master.

**Pull request merged**